File: C#Integration/PokerProtocol/Commands/Game/BetTurnStartedCommand.cs
﻿using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using EricUtility;
using PokerProtocol.Commands.Lobby.Response;
using PokerWorld.Game;

namespace PokerProtocol.Commands.Game
{
    public class BetTurnStartedCommand : AbstractCommand
    {
        protected override string CommandName
        {
            get { return BetTurnStartedCommand.COMMAND_NAME; }
        }
        public static string COMMAND_NAME = "gameBET_TURN_STARTED";

        private readonly TypeRound m_Round;
        private readonly List<int> m_CardsID;

        public TypeRound Round
        {
            get { return m_Round; }
        }

        public List<int> CardsID
        {
            get { return m_CardsID; }
        } 

        public BetTurnStartedCommand(StringTokenizer argsToken)
        {
            int count = 5;// int.Parse(argsToken.NextToken());
            m_CardsID = new List<int>();
            for (int i = 0; i < count; ++i)
            {
                m_CardsID.Add(int.Parse(argsToken.NextToken()));
            }
            m_Round = (TypeRound)int.Parse(argsToken.NextToken());
        }

        public BetTurnStartedCommand(TypeRound round,params int[] cardsID)
        {
            m_CardsID = new List<int>(cardsID);
            m_Round = round;
        }

        public BetTurnStartedCommand(TypeRound round, List<int> cardsID)
        {
            m_CardsID = cardsID;
            m_Round = round;
        }

        public override void Encode(StringBuilder sb)
        {
            //Append(sb, m_CardsID.Count);
            foreach (int ci in m_CardsID)
                Append(sb, ci);
            Append(sb, (int)m_Round);
        }
    }
}

File: JAVA/PokerProtocol/src/bluffinmuffin/protocol/commands2/game/BetTurnStartedCommand.java
package bluffinmuffin.protocol.commands2.game;

import java.util.ArrayList;
import java.util.StringTokenizer;

import bluffinmuffin.poker.entities.type.GameRoundType;
import bluffinmuffin.protocol.commands2.AbstractCommand;

public class BetTurnStartedCommand extends AbstractCommand
{
    @Override
    protected String getCommandName()
    {
        return BetTurnStartedCommand.COMMAND_NAME;
    }
    
    private final ArrayList<Integer> m_cardsID = new ArrayList<Integer>();
    private final GameRoundType m_round;
    
    public static String COMMAND_NAME = "gameBET_TURN_STARTED";
    
    public BetTurnStartedCommand(StringTokenizer argsToken)
    {
        for (int i = 0; i < 5; ++i)
        {
            m_cardsID.add(Integer.parseInt(argsToken.nextToken()));
        }
        m_round = GameRoundType.values()[Integer.parseInt(argsToken.nextToken())];
    }
    
    public BetTurnStartedCommand(Integer flop1, Integer flop2, Integer flop3, Integer turn, Integer river, GameRoundType round)
    {
        m_cardsID.add(flop1);
        m_cardsID.add(flop2);
        m_cardsID.add(flop3);
        m_cardsID.add(turn);
        m_cardsID.add(river);
        m_round = round;
    }
    
    @Override
    public void encode(StringBuilder sb)
    {
        for (int i = 0; i < 5; ++i)
        {
            append(sb, m_cardsID.get(i));
        }
        append(sb, m_round.ordinal());
    }
    
    public ArrayList<Integer> getCardsId()
    {
        return m_cardsID;
    }
    
    public GameRoundType getRound()
    {
        return m_round;
    }
}
